File: backend-for-all/backend/src/main/java/com/nexaerp/user/UserService.java
package com.nexaerp.user;

import com.nexaerp.common.response.PageResponseDto;
import com.nexaerp.user.dto.UserRequestDto;
import com.nexaerp.user.dto.UserResponseDto;
import org.springframework.web.multipart.MultipartFile;

import java.util.List;

public interface UserService {

    UserResponseDto create(UserRequestDto request);

    UserResponseDto update(Long id, UserRequestDto request);

    UserResponseDto getById(Long id);

    List<UserResponseDto> getAll();

    PageResponseDto<UserResponseDto> getAllPaged(
            int page,
            int size,
            String search,
            UserStatus status
    );

    void deactivate(Long id, String currentUserEmail);
    void activate(Long id);
}

File: accounting-system-web-project/nexa-erp-frontend/src/app/features/users/services/user.service.ts
import { HttpClient, HttpParams } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { APP_CONFIG } from '../../../core/config/app.config';
import { ApiResponse } from '../../../core/models/api-response.model';
import { PageResponse } from '../../../core/models/page.model';
import { User, UserRequest, UserStatus } from '../models/user.model';

@Injectable({
  providedIn: 'root',
})
export class UserService {
  private readonly baseUrl = `${APP_CONFIG.apiUrl}/users`;

  constructor(private http: HttpClient) {}

  getUsers(
    page = 0,
    size = 10,
    search = '',
    status?: UserStatus,
  ): Observable<ApiResponse<PageResponse<User>>> {
    let params = new HttpParams().set('page', page).set('size', size);

    if (search.trim()) {
      params = params.set('search', search.trim());
    }

    if (status) {
      params = params.set('status', status);
    }

    return this.http.get<ApiResponse<PageResponse<User>>>(this.baseUrl, { params });
  }

  getById(id: number): Observable<ApiResponse<User>> {
    return this.http.get<ApiResponse<User>>(`${this.baseUrl}/${id}`);
  }

  create(request: UserRequest): Observable<ApiResponse<User>> {
    return this.http.post<ApiResponse<User>>(this.baseUrl, request);
  }

  update(id: number, request: UserRequest): Observable<ApiResponse<User>> {
    return this.http.put<ApiResponse<User>>(`${this.baseUrl}/${id}`, request);
  }

  deactivate(id: number): Observable<ApiResponse<null>> {
    return this.http.patch<ApiResponse<null>>(`${this.baseUrl}/${id}/deactivate`, {});
  }

  activate(id: number): Observable<ApiResponse<null>> {
    return this.http.patch<ApiResponse<null>>(`${this.baseUrl}/${id}/activate`, {});
  }
}